src/data_c.c4

```diff
@@ -287,7 +287,8 @@ SHMEM_PROF_DEF_CTX_PUT_N_SIGNAL_NBI(`mem')
     SHMEM_ERR_CHECK_SYMMETRIC(target, sizeof(TYPE) * nelems);    \
     SHMEM_ERR_CHECK_NULL(source, nelems);                        \
     SHMEM_ERR_CHECK_OVERLAP(target, source, sizeof(TYPE) *       \
-                            nelems, sizeof(TYPE) * nelems, 0);   \
+                            nelems, sizeof(TYPE) * nelems, 0,    \
+                            (shmem_internal_my_pe == pe));       \
```

Comment:
Why do the checks in collectives_c get passed a 1 but these are passing the conditional? Do we know the collectives_c macros are already on the same PE?

Reply:
Good question. The "1" value is indicating the source/destination overlap check should always occur for APIs such as the collectives. However, for RMA operations, the source and destination buffers can overlap so long as the destination PE and source PE are different (i.e. `shmem_internal_my_pe == pe` evaluates to 0, ignoring the overlap check).